src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp

```diff
@@ -535,25 +480,113 @@ int XQSPIPS_Flasher::xclUpgradeFirmware(std::istream& binStream)
 
         binStream.read((char *)mWriteBuffer, size);
 
-        readFlash(addr, size);
+        readFlash(base + addr, size);
 
         mismatched = 0;
         for (unsigned i = 0; i < size; i++) {
 #if SAVE_FILE
             of_flash << mReadBuffer[i];
 #endif
-            if (mWriteBuffer[i] != mReadBuffer[i])
+            if (mWriteBuffer[i] != mReadBuffer[i]) {
                 mismatched = 1;
+                break;
+            }
+        }
+        if (mismatched) {
+            verified = false;
+            break;
         }
-        if (mismatched)
-            verify_flash.finish(false, boost::str(boost::format("Mismatch found at %d page\n") % page));
     }
-    verify_flash.finish(true, "Flash verified");
+    verify_flash.finish(verified, "Flash verified");
 
 #if SAVE_FILE
     of_flash.close();
 #endif
 
+    return mismatched;
+}
+
+int XQSPIPS_Flasher::revertToMFG(std::istream& binStream)
+{
+    initQSpiPS();
+
+    uint32_t StatusReg = XQSpiPS_GetStatusReg();
+
+    if (StatusReg == 0xFFFFFFFF) {
+        std::cout << "[ERROR]: Read PCIe device return -1. Cannot get QSPI status." << std::endl;
+        exit(-EOPNOTSUPP);
```

Comment:
should this be exit or return?

Reply:
this kind of error is not xclmgmt can handle. Just exit